File: example-projects/scala/src/test/scala/com/tngtech/jgiven/scala/example/GivenStage.scala
package com.tngtech.jgiven.scala.example

import com.tngtech.jgiven._
import com.tngtech.jgiven.annotation._

class GivenStage extends Stage[GivenStage] {

  @ProvidedScenarioState
  var someInt = 0

  @ProvidedScenarioState
  var anotherInt = 0

  def some_value(a: Int): GivenStage = {
    someInt = a
    this
  }

  def another_value(b: Int): GivenStage = {
    anotherInt = b
    this
  }

}

File: example-projects/testng/src/test/java/com/tngtech/jgiven/example/projects/testng/GivenStage.java
package com.tngtech.jgiven.example.projects.testng;

import com.tngtech.jgiven.Stage;
import com.tngtech.jgiven.annotation.ScenarioState;

class GivenStage extends Stage<GivenStage> {

    @ScenarioState
    String message;

    GivenStage message(String message) {
        this.message = message;
        return self();
    }
}
